stable/database/files/nuosm

```diff
@@ -268,6 +268,15 @@ function perform_restore() {
 
     # restore data and/or fix the metadata
     log "restoring archive and/or clearing restored archive physical metadata"
+
+    # if download is an archive, and not a backup then rename $download_dir to $DB_DIR
+    # as if download was streamed instead.
+    if [ "$restore_type" != "stream" ] && [ -f "$download_dir/1.atm" ];
+       rm -rf $DB_DIR
```

Comment:
@butson, based on our discussion today, it sounds like we expect `DB_DIR` to be empty and that we only are doing `rm -rf $DB_DIR` so that the `mv` command below does not cause the contents to be moved to a subdirectory of `DB_DIR`, e.g. `/var/opt/nuodb/archive/.../<DB name>/<DB name>`.

I think we should be more defensive about removing non-empty archive directories and change this to
```
rmdir $DB_DIR
```

This guarantees that we are not blowing away archive data accidentally and enables your use case.

Reply:
ok